Blish HUD/Controls/ColorBox.cs

```diff
@@ -1,114 +1,111 @@
-
-        //private readonly int _drawVariation;
-
-        //public ColorBox() : base() {
-        //    LoadStatics();
-
-        //    this.Size = new Point(COLOR_SIZE);
-
-        //    _drawVariation = RandomUtil.GetRandom(0, 3);
-        //}
-
-        //protected override void OnMouseMoved(MouseEventArgs e) {
-        //    base.OnMouseMoved(e);
-
-        //    this.BasicTooltipText = this.Color?.Id.ToString() ?? "None";
-        //}
-
-        //protected override CaptureType CapturesInput() {
-        //    return CaptureType.Mouse;
-        //}
-
-        //protected override void Paint(SpriteBatch spriteBatch, Rectangle bounds) {
-        //    spriteBatch.Draw(spriteBoxes[_drawVariation], bounds, this.Color?.Fur?.Rgb.ToXnaColor() ?? Microsoft.Xna.Framework.Color.White);
-
-        //    if (this.MouseOver || this.Selected)
-        //        spriteBatch.Draw(spriteHighlight, bounds, Microsoft.Xna.Framework.Color.White * 0.7f);
-        //}
-
-    //}
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ColorBox : Control {
+
+        public event EventHandler<EventArgs> OnColorChanged;
+        public event EventHandler<EventArgs> OnSelected;
+
+        private const int DEFAULT_COLOR_SIZE = 32;
+        private const string COLOR_CHANGE_SOUND_NAME = "audio/color-change";
+        private const string DRAW_VARIATION_VERSION_ONE_NAME = "colorpicker/cp-clr-v1";
+        private const string DRAW_VARIATION_VERSION_TWO_NAME = "colorpicker/cp-clr-v2";
+        private const string DRAW_VARIATION_VERSION_THREE_NAME = "colorpicker/cp-clr-v3";
+        private const string DRAW_VARIATION_VERSION_FOUR_NAME = "colorpicker/cp-clr-v4";
+        private const string HIGHLIGHT_NAME = "colorpicker/cp-clr-active";
+
+        private readonly int drawVariation;
+
+        private bool selected = false;
+
+        public bool Selected {
+            get => selected;
+            set {
+                if (SetProperty(ref selected, value)) {
+                    this.OnSelected?.Invoke(this, EventArgs.Empty);
+
+                    if (this.Visible)
+                        Content.PlaySoundEffectByName(COLOR_CHANGE_SOUND_NAME);
+                }
+            }
+        }
+
+        private Gw2Sharp.WebApi.V2.Models.Color color;
+
+        public Gw2Sharp.WebApi.V2.Models.Color Color {
+            get => color;
+            set {
+                if (color == value)
+                    return;
+
+                color = value;
+
+                OnPropertyChanged(nameof(this.ColorId));
+                OnPropertyChanged();
+                this.OnColorChanged?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        public int ColorId => this.Color?.Id ?? -1;
+
+        private static TextureRegion2D[] _possibleDrawVariations;
+        private static TextureRegion2D _spriteHighlight;
+
+        private static void LoadStatics() {
+            if (_possibleDrawVariations != null)
+                return;
```

Comment:
This shouldn't be possible since it is set in this call and the static constructor will never get called again anyways.  I think it's another relic of my old code.

Reply:
It's not in a static constructor. It gets called everytime a ColorBox gets initiated. I could change the `LoadStatic` to a static contructor though

Reply:
Oh, you're absolutely right.  I'm so used to them all being in a static ctor now that I didn't even notice.